velox/jni/java/velox-jni-core/src/main/java/velox/jni/NativePlanBuilder.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package velox.jni;
+
+import org.apache.spark.sql.types.StructType;
+
+public class NativePlanBuilder extends NativeClass {
+  public NativePlanBuilder() {
+    setHandle(nativeCreate());
+  }
+
+  private native void nativeJavaScan(String schema);
+
+  private native void nativeFilter(String condition);
+
+  private native void nativeProject(String[] projections);
+
+
+  private native String nativeBuilder();
+
+  protected native long nativeCreate();
+
+  private native void nativeRelease();
+
+  private native String nativeNodeId();
```

Comment:
If we were to record all operations, that could work too, but this would make the process more complicated. And would it make the errors more apparent? I believe that the error should be thrown during the .velox.project phase, not during the velox.project().filter.build phase when building, right?